skimage/util/lookfor.py

```diff
@@ -1,7 +1,10 @@
-import numpy as np
 import sys
 
+from .._shared.utils import deprecate_func
+from .._vendored.numpy_lookfor import lookfor as _lookfor
```

Comment:
Highlighting this. Unfortunately we relied on NumPy's lookfor which is promptly removed in NumPy 2.0. Technically, if we don't want to make a breaking release we need to deprecate the function in a deprecation cycle for which we need to vendor NumPy's lookfor.

Though, skimage.lookfor is an interactive function, so we might get away with not vendoring NumPy's lookfor, and just making it return a deprecation warning...

Reply:
If its annoying, I guess we could also vendor it back.  Does anyone still use lookfor?  I feel as an interactive function, it would be a few old-school power users probably (I never used it), thus the feeling we probably can get away with (and no actual program can fail).

Reply:
Fair, I am not sure if some IDE has search features, which would be nice.  It still seems odd in NumPy to me and probably very rarely used, but if you prefer keeping it for now, I am fine with that too.

Reply:
I think most IDEs only take the function / method / class name into account when matching for a search string. Or you can do a full search that includes more than just docstrings. @seberg do you have an opinion on how much maintenance work this has been? 

Of all discussed options, I prefer NumPy to just keep it around as that's the most simple option from our side. However, I'd totally get why you might want to remove `lookfor`. In that case I'd vote for the scientific python package option or removing it all-together. I intend vendoring to be temporary solution only.

Reply:
I am not sure it has been maintanence work, so if there is a PR to restore it I won't mind merging it.
To me it was mostly about namespace clutter as I don't think typical user use it.  IIRC it also returns some nonsense results (i.e. things that are much better in the html docs, but it doesn't find those, while it found docs that belonged just deleted).

Reply:
Realizing @Carreau may also have thought about this problem.

Reply:
I don't have any particular solution. 
I plan to have full documentation search with papyri at some point, we can even do some indexing. but it's not ready yet.

Reply:
Then let's keep it until the status quo changes.